lib/graphql/fragment_cache/schema/lazy_cache_resolver.rb

```diff
@@ -0,0 +1,41 @@
+require "graphql/fragment_cache/fragment"
+
+module GraphQL
+  module FragmentCache
+    module Schema
+      using Ext
+      class LazyCacheResolver
+        def initialize(fragment, query_ctx, object_to_cache, &block)
+          @fragment = fragment
+          @query_ctx = query_ctx
+          @object_to_cache = object_to_cache
+          @lazy_state = query_ctx[:lazy_cache_resolver_statez] ||= {
+            pending_fragments: Set.new,
+            resolved_fragments: {}
+          }
+          @block = block
+
+          @lazy_state[:pending_fragments] << @fragment
+        end
+
+        def resolve
+          unless @lazy_state[:resolved_fragments].key?(@fragment)
+            resolved_fragments = Fragment.read_multi(@lazy_state[:pending_fragments].to_a)
```

Comment:
You mentioned that you see `write_multi` but also multiple `get` calls. Does it work for you now? If not, can you check what's inside `@lazy_state` in such cases?

Reply:
I was going to see whether Redis' `read_multi` was being used, but I think I got carried away and ended up looking into how Cacher does persisting and forgot about `read_multi` 😅.

The behavior I was getting was like this, for some reason I would see first `MGET` with correct cache keys, but weirdly concatenated.
<img width="936" alt="Screen Shot 2021-06-28 at 8 42 44 PM" src="https://user-images.githubusercontent.com/26189247/123656055-6433d080-d851-11eb-8539-17c51ce8f3aa.png">

The problem was that I passed an array of cache keys to `read_multi(*keys)` when I should have splatted the array. I've fixed that in this [commit](https://github.com/DmitryTsepelev/graphql-ruby-fragment_cache/pull/70/commits/c241ef60f1812f54263d31490dda071dcabdde92).

After that, logs look like this:

<img width="945" alt="Screen Shot 2021-06-28 at 8 46 19 PM" src="https://user-images.githubusercontent.com/26189247/123656602-e4f2cc80-d851-11eb-8faa-65168a098ee3.png">

plus, the query response now feels significantly faster.